server/sonar-server/src/main/java/org/sonar/server/user/ws/SetHomepageAction.java

```diff
@@ -51,9 +73,47 @@ public void define(WebService.NewController controller) {
 
   @Override
   public void handle(Request request, Response response) throws Exception {
+    userSession.checkLoggedIn();
 
-    // WIP : Not implemented yet.
-    response.noContent();
+    String type = request.mandatoryParam(PARAM_TYPE);
+    String value = request.param(PARAM_VALUE);
+
+    if (PROJECT.toString().equals(type) && isBlank(value)) {
```

Comment:
suggestion: inline with `checkArgument`

Reply:
not that intuitive with the double check but done